File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s910447421_commented.py
class FastScanner:
    def __init__(self):
        self.buffer = []
        self.reader = sys.stdin

    def next(self):
        while not self.buffer:
            for token in self.reader.readline().split():
                self.buffer.append(token)
        return self.buffer.pop(0)

    def next_int(self):
        return int(self.next())

    def next_long(self):
        return int(self.next())

    def next_double(self):
        return float(self.next())


class OutPut:
    def __init__(self):
        self.writer = sys.stdout

    def pln(self, x):
        print(x, file=self.writer)
        self.writer.flush()

    def p(self, x):
        print(x, end='', file=self.writer)
        self.writer.flush()


def gcd(a, b):
    if a == 0 or b == 0:
        return max(a, b)
    return gcd(min(a, b), max(a, b) % min(a, b))


def fast_exp(base, exp, mod):
    ans = 1
    while exp > 0:
        if exp % 2 == 1:
            ans *= base
        exp //= 2
        base *= base
        base %= mod
        ans %= mod
    return ans


def mod_inv(num, mod):
    return fast_exp(num, mod - 2, mod)


def pop(x):
    cnt = 0
    while x > 0:
        if x % 2 == 1:
            cnt += 1
        x //= 2
    return cnt


def min(a, b):
    return math.min(a, b)


def max(a, b):
    return math.max(a, b)


def abs(x):
    return math.abs(x)


def ceil(num, den):
    ans = num // den
    if num % den != 0:
        ans += 1
    return ans


def solve():
    global ans
    for add in range(1, N):
        taken = [False] * N
        p = balls[add][0] - balls[0][0]
        q = balls[add][1] - balls[0][1]
        ans = min(ans, check(p, q))


def check(p, q):
    ret = 0
    while True:
        added = False
        start = 0
        x = -1
        y = -1
        for i in range(N):
            if not taken[i]:
                added = True
                ret += 1
                start = i
                taken[i] = True
                x = balls[i][0] + p
                y = balls[i][1] + q
                break
        if not added:
            break
        for i in range(start + 1, N):
            if balls[i][0] == x and balls[i][1] == y:
                taken[i] = True
                x += p
                y += q
    return ret


if __name__ == "__main__":
    import sys
    import math

    ans = 100
    balls = []
    N = 0
    taken = []

    I = FastScanner()
    O = OutPut()
    N = I.next_int()
    ans = N
    balls = [[I.next_int(), I.next_int()] for _ in range(N)]
    balls.sort(key=lambda x: x[0])
    solve()
    for i in range(N):
        balls[i][0], balls[i][1] = balls[i][1], balls[i][0]
    balls.sort(key=lambda x: x[0])
    solve()
    O.pln(ans)  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/Go/s910447421_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

var ans = 100
var balls [][2]int
var N int
var taken []bool

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	N = readInt(reader)
	ans = N
	balls = make([][2]int, N)
	for i := 0; i < N; i++ {
		balls[i][0] = readInt(reader)
		balls[i][1] = readInt(reader)
	}
	sort.Slice(balls, func(i, j int) bool { return balls[i][0] < balls[j][0] })
	solve()
	for i := range balls {
		x := balls[i][0]
		y := balls[i][1]
		balls[i][0] = y
		balls[i][1] = x
	}
	sort.Slice(balls, func(i, j int) bool { return balls[i][0] < balls[j][0] })
	solve()
	fmt.Fprintln(writer, ans)
}

func solve() {
	for add := 1; add < N; add++ {
		taken = make([]bool, N)
		p := balls[add][0] - balls[0][0]
		q := balls[add][1] - balls[0][1]
		ans = min(ans, check(p, q))
	}
}

func check(p, q int) int {
	ret := 0
	for {
		added := false
		start := 0
		var x, y int
		for i := 0; i < N; i++ {
			if !taken[i] {
				added = true
				ret++
				start = i
				taken[i] = true
				x = balls[i][0] + p
				y = balls[i][1] + q
				break
			}
		}
		if !added {
			break
		}
		for i := start + 1; i < N; i++ {
			if balls[i][0] == x && balls[i][1] == y {
				taken[i] = true
				x += p
				y += q
			}
		}
	}
	return ret
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func readInt(reader *bufio.Reader) int {
	str, _ := reader.ReadString('\n')
	str = str[:len(str)-1]
	num, _ := fmt.Sscanf(str, "%d", new(int))
	return num
}

// End of Code
